aws/signer/v4/middleware_test.go

```diff
@@ -124,10 +125,11 @@ func TestSignHTTPRequestMiddleware(t *testing.T) {
 				return out, metadata, err
 			})
 
-			ctx := SetSigningMetadata(context.Background(), SigningMetadata{
-				SigningName:   signingName,
-				SigningRegion: signingRegion,
-			})
+			ctx := func() context.Context {
+				ctx := awsmiddleware.SetSigningName(context.Background(), signingName)
+				ctx = awsmiddleware.SetSigningRegion(ctx, signingRegion)
+				return ctx
+			}()
```

Comment:
Why the closure?

Reply:
I felt that it made the code cleaner to read, but if there is confusion around the closure then I will just inline everything.

Reply:
Not a big deal since a part of a test and not runtime. It surprised me and had to read a few times to figure out if i was missing something